File: Core/actor/valueactor.cpp
#include "valueactor.h"

ValueActor::ValueActor(ValueGroup *valueGroup, QString id, VALUE_TYPE valueType, QVariant::Type typeHint, QObject *parent)
    : ActorBase(valueGroup, id, valueType, typeHint, parent)
{

}

bool ValueActor::cmdSupported(ACTOR_CMDS cmd) {
    return cmd == ACTOR_CMDS::ACTOR_CMD_SET_VALUE;
}

bool ValueActor::isAsync() {
    return true;
}

void ValueActor::_triggerCmd(ACTOR_CMDS cmd) {
    if (cmd == ACTOR_CMDS::ACTOR_CMD_SET_VALUE) {
        Q_EMIT(requestSetValue());
    }
}

QVariant ValueActor::_updateValue(QVariant newValue) {
    if (newValue.canConvert(m_typeHint)) {
        newValue.convert(m_typeHint);
        return newValue;
    } else {
        qWarning() << "Cannot convert" << newValue << "to type" << m_typeHint;
        return QVariant::fromValue(newValue);
    }
}

File: AndroidApp/OshCore/src/main/java/com/osh/actor/ValueActor.java
package com.osh.actor;

import com.osh.value.ValueGroup;
import com.osh.value.ValueType;

public class ValueActor extends ActorBase {

    @Override
    protected Object _updateValue(Object newValue) {
        return newValue;
    }

    public ValueActor(ValueGroup valueGroup, String id, ValueType valueType) {
        super(valueGroup, id, valueType);
    }

    @Override
    public boolean cmdSupported(ActorCmds cmd) {
        return cmd == ActorCmds.ACTOR_CMD_SET_VALUE;
    }

    @Override
    protected void _triggerCmd(ActorCmds cmd) {

    }



}
